sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FoldablePropagationSuite.scala

```diff
@@ -130,6 +130,20 @@ class FoldablePropagationSuite extends PlanTest {
     comparePlans(optimized, correctAnswer)
   }
 
+  test("Propagate in inner join") {
+    val ta = testRelation.select('a, Literal("a").as('tag))
+      .union(testRelation.select('a, Literal("b").as('tag)))
+      .subquery('ta)
+    val tb = testRelation.select('a, Literal("a").as('tag))
+      .union(testRelation.select('a, Literal("b").as('tag)))
+      .subquery('tb)
+    val query = ta.join(tb, Inner,
+      Some("ta.a".attr === "tb.a".attr && "ta.tag".attr === "tb.tag"))
```

Comment:
-> `Some("ta.a".attr === "tb.a".attr && "ta.tag".attr === "tb.tag".attr))`

Then add the rule [`ConstantFolding` into the test suite.](https://github.com/stanzhai/spark/blob/15fae5029e7c45d3b2a3108e24ca235a6a6fccfc/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FoldablePropagationSuite.scala#L31)